### DISCUSSION TITLE
AbsoluteLayout not working

### DISCUSSION THREAD

**Discussion by ac2021-June** [unoplatform/uno#7817] [Q&A]
### Current behavior

Nothing gets render.

<AbsoluteLayout>
<Grid HorizontalAlignment="Stretch" VerticalAlignment="Stretch" AbsoluteLayout.LayoutBounds="10,0">
<TextBlock Text="Text1"/>
<TextBlock Text="Text1"/>
</Grid>
<Grid HorizontalAlignment="Stretch" VerticalAlignment="Stretch" AbsoluteLayout.LayoutBounds="0,0">
<TextBlock Text="Text1"/>
<TextBlock Text="Text1"/>
</Grid>
 </AbsoluteLayout>

### Expected behavior

The 2 grids, should get rendered at the given Left, top coordinates. 

### Affected platforms

iOS, Android

### IDE

Visual Studio 2019

**Comment by jeromelaban** [comment]
Thanks for the report. Can you add a repro sample so we can make sure we're discussing the same issue?

**Comment by ac2021-June** [comment]
Please find the code at the location: 
https://github.com/ac2021-June/AbsoluteLayoutSample.git

also, is there a way to stretch the elements present on the Canvas control? This doesnt work.. Trying to figuring out a way to position my controls in the absolute position. 
`<Grid x:Name="gd">
        <Canvas Width="{Binding ElementName=gd, Path=ActualWidth}"
                Height="{Binding ElementName=gd, Path=ActualHeight}">
            <Grid Canvas.Left="0"
                  Canvas.Top="100">
                <Grid.RowDefinitions>
                    <RowDefinition Height="*" />
                    <RowDefinition Height="*" />
                </Grid.RowDefinitions>
                <Rectangle Fill="Red" />
                <Rectangle Fill="Black" />
            </Grid>
        </Canvas>
    </Grid>
`

<blockquote><img src="https://opengraph.githubassets.com/bb0f4f83038a73973c4e878563ce898700fd369b80053ec14a6bf00fb1273736/ac2021-June/AbsoluteLayoutSample" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/ac2021-June/AbsoluteLayoutSample">GitHub - ac2021-June/AbsoluteLayoutSample</a></strong></div><div>Contribute to ac2021-June/AbsoluteLayoutSample development by creating an account on GitHub.</div></blockquote>

**Reply by jeromelaban** [reply]
Thanks. `AbsoluteLayout` is a Xamarin Forms control, not a WinUI/Uno control, so will not be able to use it.

If you want to place controls in a canvas, take a look at [this sample](https://playground.platform.uno/#canvas) and the [`Canvas` documentation](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.canvas?view=winrt-22000).

**Reply by ac2021-June** [reply]
Sample link not opening.. also, the Canvas control is not taking the parent's width and height as per the documentation,

**Reply by ac2021-June** [reply]
```xaml
<Grid x:Name="gd"> 
<Canvas Width="{Binding ElementName=gd, Path=ActualWidth}" Height="{Binding ElementName=gd, Path=ActualHeight}"> 
<Grid Canvas.Left="0" Canvas.Top="100">
 <Grid.RowDefinitions> 
<RowDefinition Height="*" /> 
<RowDefinition Height="*" /> 
</Grid.RowDefinitions> 
<Rectangle Fill="Red" /> 
<Rectangle Fill="Black" /> 
</Grid> 
</Canvas> 
</Grid>
```

This code doesnt work for the Canvas width and height... Please let us know, what is wrong above.

**Reply by ac2021-June** [reply]
we want to make Canvas's height and width same as its parent control Grid.

**Reply by jeromelaban** [reply]
ActualWith and ActualHeight are not properties that are meant to be data bound, as they are not dependency properties. If displays something, it'll likely be one step behind the proper values at every layout update cycle.

Make sure to get your sample working on windows first, either using the UWP or WinUI heads, or by using [XAML studio](https://www.microsoft.com/store/productId/9NTLS214TKMQ). If the behavior on Uno's targets is different from what you find on Windows, please let us know.

Also, if https://playground.platform.uno/#d0bf291d does not open at all, you may have additional issues that we're interested in troubleshooting.

**Reply by ac2021-June** [reply]
We are getting following error, while trying to access the above application. Please provide it on github so that we can download and have a look 
![image](https://user-images.githubusercontent.com/85895535/149539849-cc23dd91-aa6b-411c-90fa-ee0e6a2fee97.png)

**Reply by jeromelaban** [reply]
Thanks, this link should be working, if you can profile the output of the browser log, it'll be helpful.

The XAML in the sample does not matter much as it's not doing what you're expecting it to do, but here it is:
```xaml
<Grid x:Name="gd"> 
    <Canvas Width="{Binding ElementName=gd, Path=ActualWidth}" 
            Height="{Binding ElementName=gd, Path=ActualHeight}"> 
        <Grid Canvas.Left="0" Canvas.Top="100">
            <Grid.RowDefinitions> 
                <RowDefinition Height="*" /> 
                <RowDefinition Height="*" /> 
            </Grid.RowDefinitions> 
            <Rectangle Grid.Row="0" Fill="Red" Width="100" Height="100" /> 
            <Rectangle Grid.Row="1" Fill="Black" Width="100" Height="100" /> 
        </Grid>
    </Canvas>
</Grid>
```
